File: 230420/二分探索/typical90-007.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

constexpr long long INF = (1LL << 60);

int main() {
    int n, q;
    cin >> n;
    std::vector<long long> a(n);
    for (auto &x : a) cin >> x;

    a.push_back(INF);
    a.push_back(-INF);
    sort(a.begin(), a.end());

    cin >> q;
    for (int i = 0; i < q; i++) {
        long long b;
        cin >> b;
        auto iter = lower_bound(a.begin(), a.end(), b);
        long long val1 = b - *prev(iter);
        long long val2 = *iter - b;
        cout << min(val1, val2) << endl;
    }
    return 0;
}

File: 230420/二分探索/typical90-007.py
INF = 1 << 60

n = int(input())
a = list(map(int, input().split()))
q = int(input())

a.append(INF)
a.append(-INF)
a.sort()

for i in range(q):
    b = int(input())
    left, right = 0, n + 1
    while abs(left - right) > 1:
        mid = (left + right) // 2
        if a[mid] > b:
            right = mid
        else:
            left = mid
    val1 = a[right] - b
    val2 = b - a[left]
    print(min(val1, val2))